### PR TITLE
[JENKINS-58391] Add Distribution Group parameter

### PR DESCRIPTION
**Pull request by Sirrah**
This is my attempt at supporting the Distribution Group parameter:

https://issues.jenkins-ci.org/browse/JENKINS-58391

I'm not familiar with Jenkins plugins so I'd appreciate any feedback on whether this is the right direction and if anything else needs to be done.

### CODE REVIEW

**Comment by mezpahlan** [comment]
Wow! Absolutely top job! Many many thanks for this. Let me do a few little checks but I'm very happy with this so far. And you've even got tests!

File: src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java [hunk 1]
```diff
@@ -141,6 +146,7 @@ public FormValidation doCheckOwnerName(@QueryParameter String value) {
                 return FormValidation.error(Messages.AppCenterRecorder_DescriptorImpl_errors_missingOwnerName());
             }
 
+            // TODO proper validator, every non-empty string should be valid?
```
Comment (mezpahlan) [inline src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java]: If you try and change your username in the AppCenter UI on their website and enter, for example, a `:` character then the error tells you that:

>Username can only contain letters, numbers, dashes, underscores, or periods.

So I would remove this TODO.

File: src/main/resources/io/jenkins/plugins/appcenter/Messages.properties [hunk 2]
```diff
@@ -1,9 +1,11 @@
 AppCenterRecorder.DescriptorImpl.errors.missingApiToken=Please specify an AppCenter API Token
-AppCenterRecorder.DescriptorImpl.errors.invalidApiToken=API Token canot contain whitespace
+AppCenterRecorder.DescriptorImpl.errors.invalidApiToken=API Token can't contain whitespace
 AppCenterRecorder.DescriptorImpl.errors.missingOwnerName=Please specify an AppCenter Owner Name
 AppCenterRecorder.DescriptorImpl.errors.invalidOwnerName=Username can only contain letters, numbers, dashes, underscores, or full stops
 AppCenterRecorder.DescriptorImpl.errors.missingAppName=Please specify an AppCenter App Name
 AppCenterRecorder.DescriptorImpl.errors.invalidAppName=App name cannot contain whitespace
+AppCenterRecorder.DescriptorImpl.errors.missingDistributionGroup=Please specify a Distribution Group
+AppCenterRecorder.DescriptorImpl.errors.invalidDistributionGroup=Distribution Group can only contain letters, numbers, dashes, underscores, or spaces
```
Comment (mezpahlan) [inline src/main/resources/io/jenkins/plugins/appcenter/Messages.properties]: I was able to create the distribution group: `:feahifuh[{}[}{{}:\|``¬¬¬¬!"£!"%£^$&%*^(&)` via the AppCenter UI.

So perhaps we do need to support more than what this suggests?

File: src/main/resources/io/jenkins/plugins/appcenter/Messages.properties [hunk 3]
```diff
@@ -1,9 +1,11 @@
 AppCenterRecorder.DescriptorImpl.errors.missingApiToken=Please specify an AppCenter API Token
-AppCenterRecorder.DescriptorImpl.errors.invalidApiToken=API Token canot contain whitespace
+AppCenterRecorder.DescriptorImpl.errors.invalidApiToken=API Token can't contain whitespace
```
Comment (mezpahlan) [inline src/main/resources/io/jenkins/plugins/appcenter/Messages.properties]: You're correct, this was a typo on my end but could you change it to: `cannot` please. Sorry, I know it is minor, but that was what I was intending.

File: src/main/resources/io/jenkins/plugins/appcenter/Messages.properties (hunk 2 above)
Comment (Sirrah) [reply src/main/resources/io/jenkins/plugins/appcenter/Messages.properties]: Probably. I'd suggest generalising this message and leave the related validator as it is. That leaves us to check if the distribution group value is present and leave the actual validation to AppCenter.

File: src/main/resources/io/jenkins/plugins/appcenter/Messages.properties (hunk 3 above)
Comment (Sirrah) [reply src/main/resources/io/jenkins/plugins/appcenter/Messages.properties]: Sure 👍

File: src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java (hunk 1 above)
Comment (Sirrah) [reply src/main/java/io/jenkins/plugins/appcenter/AppCenterRecorder.java]: Yeah, that should be removed. It's a left-over from before I put in the `DistributionGroupValidator`.

**Comment by Sirrah** [comment]
@mezpahlan Thanks! I've updated the PR based on your comments. I don't think there is much we can or should do in `DistributionGroupValidator`. As you noticed, AppCenter allows a wide range of names.

**Pull request merged**